src/binary/runBinary.ts

```diff
@@ -13,14 +14,17 @@ export default async function runBinary(
 
   const context = getTabnineExtensionContext();
   const args: string[] = [
-    "--client=vscode",
+    ON_PREM ? "--client=vscode_on_prem" : "--client=vscode",
     "--no-lsp=true",
     tabnineExtensionProperties.logFilePath
       ? `--log-file-path=${tabnineExtensionProperties.logFilePath}`
       : null,
+    ON_PREM ? "--onprem=true" : null,
+    ON_PREM ? "--no_bootstrap" : null,
     tabnineExtensionProperties.logLevel
       ? `--log-level=${tabnineExtensionProperties.logLevel}`
       : null,
+    tabnineExtensionProperties.cloudHost ? `--cloud2_url=${tabnineExtensionProperties.cloudHost}` : null,
```

Comment:
Pass --onprem only if that set?

Reply:
the other way around otherwise it won't work in regular mode. fixed that